### ISSUE TITLE
Sequential reduction without initial value

### ISSUE THREAD

**Issue by jdgarciauc3m** [arcosuc3m/grppi#200]
When no initial value is provided to reduce, the initial value may be computed from the first two elements in a sequence.

~~~c++
auto init = combine_op(first, first+1)
return reduce(ex, first+2, last, init, combine_op);
~~~

Note that reductions for sequences shorter than 2 are non-sense for reductions without initial value.

**EDIT**: Actually for a sequence of size 1 (distance(first,last)==1), the result of the reduction should be *first

**Comment by Drioastorga** [comment]
We decided to do not support the reduction without initial value .

**Comment by jdgarciauc3m** [comment]
Solved by PR #203
